pkg/controllers/tagging/tagging_controller.go

```diff
@@ -31,19 +31,30 @@ import (
 
 // workItem contains the node and an action for that node
 type workItem struct {
-	node   *v1.Node
-	action func(node *v1.Node) error
+	node           *v1.Node
+	action         func(node *v1.Node) error
+	requeuingCount int
+	enqueueTime    time.Time
 }
 
+const (
+	maxRequeuingCount           = 9
+	totalErrors                 = "total_errors"
+	workItemProcessingTime      = "work_item_processing_time"
+	workItemDequeuingTime       = "work_item_dequeuing_time"
+	errorsAfterRetriesExhausted = "errors_after_retries_exhausted"
```

Comment:
Why does this need to be a separate metric from `total_errors`?

Reply:
It's because we want to know the number of error a particular item encountered before failing out in this case, while the `totalErrors` is for the number of errors before succeeding. I think it's gonna be easier to query the metrics later on if we separate them like so. Pls let me know your recommendations.

Reply:
I think what you are saying makes sense--you wouldn't be able to tell the how many errors were seen before a success otherwise (I don't think?)